src/app/component/CategoryProductList/CategoryProductList.component.js

```diff
@@ -22,154 +24,236 @@ import './CategoryProductList.style';
+
+        if (isUpdateable && shouldUpdateList && !isLoading) {
+            const modifier = next ? maxPage + 1 : minPage - 1;
+
+            this.setState({ pagesCount: pagesCount + 1 });
+            loadPage(modifier);
         }
+    }
 
-        if (showLoadMore) {
-            return (
+    renderLoadButton(next = false, hidden = false) {
+        const loadPage = this.loadPage.bind(null, next);
+        return (
+            <div
+              block="CategoryProductList"
+              elem="LoadButton"
+              mods={ { hidden } }
+              role="button"
+              tabIndex="-1"
+              onKeyUp={ loadPage }
+              onClick={ loadPage }
+            >
+                    { next ? __('Load next') : __('Load previous') }
+            </div>
+        );
+    }
+
+    renderNoProducts() {
+        return (
+            <div
+              block="CategoryProductList"
+            >
                 <div
                   block="CategoryProductList"
-                  elem="Placeholder"
-                  ref={ (node) => { this.node = node; } }
+                  elem="NoProducts"
                 >
-                    { renderPlaceholders() }
+                    No products found
                 </div>
-            );
-        }
+            </div>
+        );
+    }
 
-        return null;
+    renderPage(items, key) {
+        const { customFilters, isLoading } = this.props;
+
+        return (
+            <ul
+              block="CategoryProductList"
+              elem="Page"
+              key={ key }
+              mods={ { isLoading } }
+              ref={ (node) => { this.nodes[key] = node; } }
+            >
+                { items.map(product => (
+                    <ProductCard
+                      product={ product }
+                      key={ product.id }
+                      customFilters={ customFilters }
+                      arePlaceholdersShown
+                    />
+                )) }
+            </ul>
+        );
     }
 
     render() {
-        const { items, totalItems, isLoading } = this.props;
-        const showLoadMore = items.length < totalItems && !isLoading;
+        const { pages, totalPages, isLoading } = this.props;
+        const { minPage, maxPage } = this.getPagesBounds();
+
+        const showLoadPrevious = minPage > 1 && !isLoading;
+        const showLoadNext = maxPage < totalPages;
+
+        if (!isLoading && totalPages === 0) return this.renderNoProducts();
 
         return (
-            <ul block="CategoryProductList" mods={ { isLoading } }>
-                { !isLoading && this.renderProducts() }
-                { this.renderPlaceholder(showLoadMore, isLoading) }
-            </ul>
+            <div block="CategoryProductList">
+                { showLoadPrevious && this.renderLoadButton() }
+                { !isLoading && Object.entries(pages).map(([pageNumber, items]) => this.renderPage(items, pageNumber)) }
```

Comment:
Move this to renderPgaes function, do not pass any props, make that function independent from main render and responsible for internal content display. Check if it looks better :)